source/mqtt_agent.c

```diff
@@ -796,6 +796,76 @@ static void clearPendingAcknowledgments( MQTTAgentContext_t * pMqttAgentContext
 
 /*-----------------------------------------------------------*/
 
+static bool validateStruct( MQTTAgentContext_t * pMqttAgentContext,
+                            CommandInfo_t * pCommandInfo )
+{
+    bool ret = false;
+
+    if( ( pMqttAgentContext == NULL ) ||
+        ( pCommandInfo == NULL ) )
+    {
+        LogError( ( "Pointer cannot be NULL. pMqttAgentContext=%p, pCommandInfo=%p.",
+                    ( void * ) pMqttAgentContext,
+                    ( void * ) pCommandInfo ) );
+        ret = false;
```

Comment:
This line can be removed.

Reply:
That's true, but I didn't know how MISRA would react since the block would be empty otherwise (since logs are compiled out). I removed it for now and can add it back if it causes an issue